server/src/main/java/com/vaadin/server/VaadinService.java

```diff
@@ -674,7 +676,7 @@ private VaadinSession findOrCreateVaadinSession(VaadinRequest request)
             return doFindOrCreateVaadinSession(request,
                     requestCanCreateSession);
         } finally {
-            unlockSession(wrappedSession);
```

Comment:
Should the now unused `unlockSession` method be removed or deprecated? Removing would be more in line with the approach taken with `lockSession`, i.e. to break it properly.

Another alternative would be to keep it as `unlockSession(WrappedSession, Lock)` so that a totally custom implementation might still override the way unlocking happens in a practical way.

Reply:
Good pick. It did not notice that findOrCreateVaadinSession was the sole user of unlockSession.

Reply:
Shouldn't that new method be used here as well? Otherwise, it's quite pointless to change the signature and the implementation.